### DISCUSSION TITLE
Can someone help me with the conversion of these scripts?

### DISCUSSION THREAD

**Discussion by tonysprenk** [RavenSystem/esp-homekit-devices#2094] [MEPLHAA Scripts]
I have a few devices that I converted to the new scripts and most of them are working but I’m getting stuck on a few. Any help would be greatly appreciated. 

This is a smart outlet with a button to turn it on or off. Everything is working except the button does nothing now. 

Old script: 

``` 
{"c":{"cd":5,"b":[{"g":13,"t":5}]},"a":[{"t":2,"0":{"r":[{"g":5},{"g":12}]},"1":{"r":[{"g":5,"v":1},{"g":12,"v":1}]},"b":[{"g":13}]}]}
``` 

New script: 

```
{"c":{"io":[[[5,4],2],[[13],6]],"l":4,"b":[[13,5]]},"a":[{"t":2,"0":{"r":[[5]]},"1":{"r":[[5,1]]},"b":[[13]]}]}
``` 

This is a lock mechanism, the lock is not triggered when I lock it in the home app: 

old:
``` 
{"a":[{"t":4,"0":{"r":[{"g":5,"v":1}]},"1":{"r":[{"g":5}]}},{"t":1}]}
``` 
New:
``` 
{"a":[{"0":{"r":[[5,1]]},"1":{"r":[[5]]},"t":4},{"t":1}],"c":{"io":[[[5],2]]}}
``` 

This is an IR remote controlling a fan, the fan is not triggered by any of the actions:

old:
``` 
{"c":{"n":"Fan","t":14,"p":"U5J%AaDTAaAaAa"},"a":[{"t":65,"0":{"i":[{"c":"aHiHhHf"}]},"1":{"i":[{"c":"aHiHhHf"}]}},{"t":1,"0":{"i":[{"c":"aHiBaEbAeBaEbAc"}]},"1":{"i":[{"c":"aHiBaEbAeBaEbAc"}]}},{"t":1,"i":0.5,"1":{"i":[{"c":"aHjHhHe"}]}}]}
``` 
New:
```
{"c":{"n":"Fan","t":14,"p":"U5J%AaDTAaAaAa","io":[]},"a":[{"0":{"i":[{"c":"aHiHhHf"}]},"1":{"i":[{"c":"aHiHhHf"}]},"t":65},{"0":{"i":[{"c":"aHiBaEbAeBaEbAc"}]},"1":{"i":[{"c":"aHiBaEbAeBaEbAc"}]},"t":1},{"1":{"i":[{"c":"aHjHhHe"}]},"t":1,"i":0.5}]}
``` 

This is a doorbell, the doorbell sound on the HomePod is not triggered when the button is pressed:

old:
``` 
{"a":[{"t":13,"f0":[{"g":14,"t":1}]}]}
``` 

new:
``` 
{"a":[{"t":13,"f0":[[14,1]]}],"c":{"io":[[[14],6]]}}
``` 

Thanks again for any help!

**Comment by Tellicious** [comment]
Just a few things that I saw at a glance:
- The two scripts for the smart outlet are significantly different: the first one is using GPIO5 and 12, there is a creation delay and no LED. the second one uses only GPIO5, there is a LED connected to GPIO4 and no creation delay.
The v12 script same as the first one would be:
`{"c":{"io":[[[5,12],2],[[13],6]],"b":[[13,5]],"cd":5},"a":[{"t":2,"0":{"r":[[5],[12]]},"1":{"r":[[5,1],[12,1]]},"b":[[13]],"s":0}]}`
**Watch-out that in v12 internal pull-up and pull-down resistors have to be manually configured, as the pin is floating by default** [Link](https://github.com/RavenSystem/esp-homekit-devices/wiki/gpios-configuration#mode-6-input-only-with-binary-input-buttonswitch-support)
If there are no HW pull-ups in your board, you could try this:
`{"c":{"io":[[[5,12],2],[[13],6,1,1]],"b":[[13,5]],"cd":5},"a":[{"t":2,"0":{"r":[[5],[12]]},"1":{"r":[[5,1],[12,1]]},"b":[[13]],"s":0}]}`
- For the lock, have you tried moving `"c"` before `"a"`?
- For the doorbell, again the pull-up thing might still apply
